src/App.js

```diff
@@ -4,29 +4,37 @@ import "./App.css";
 import * as yup from "yup";
 
 const schema = yup.object().shape({
-  picture: yup
-    .mixed()
+  picture: yup.mixed()
     .required("You need to provide a file")
-    .test("fileSize", "The file is too large", (value) => {
-      return value && value[0].size <= 2000000;
+    .test("fileSize", "The file is too large", function(value) {
+      console.log('*',value)
+      return false
```

Comment:
I've tried to force the validation rule to return a validation error.

This resulted in an exception being thrown:
<img width="1664" alt="Screen Shot 2021-11-03 at 13 55 40" src="https://user-images.githubusercontent.com/450319/140063813-52fa67a2-ab53-4858-9351-1e9a03570e3f.png">


🤷‍♂️ 

I suspect that this might be a bug in React Hook Form. So I plan to update it.